com.ibm.streamsx.topology/opt/python/packages/streamsx/rest_primitives.py

```diff
@@ -1387,14 +1403,41 @@ def __init__(self, rest_client, credentials):
         self.rest_client = rest_client
         self._credentials = credentials
         self._v2_rest_url = self._credentials[_IAMConstants.V2_REST_URL]
+        self._jobs_url = None
+
+    def _get_jobs_url(self):
+        """Get & save jobs URL from the status call."""
+        if self._jobs_url is None:
+            self.get_instance_status()
```

Comment:
I was confused when I read this initially since`get_instance_status`now has the side effect of getting the jobs url. Maybe there could be something like a `seturls` method which gets called in the delegator constructor.

Reply:
I'm trying to avoid an extra URL request to get URLs that might never be used.